### PR TITLE
FEAT: Allow redirectedurls to apply to assets

### PR DESCRIPTION
**Pull request by madmatt**
This only applies if the `silverstripe/assets` module is installed, we don't require it for this module.

Note: Requires silverstripe/silverstripe-assets/pull/391 to be merged prior to merging this. As such we probably shouldn't merge this until that has been merged and a new stable released (so a new stable of this module can be released to coincide with it - although I don't think you can specify 'minimum version of a suggested module', right?)

### CODE REVIEW

File: src/Service/RedirectedURLService.php [hunk 1]
```diff
@@ -0,0 +1,133 @@
...
+use SilverStripe\Control\HTTPRequest;
+use SilverStripe\Control\HTTPResponse;
+use SilverStripe\Control\HTTPResponse_Exception;
+use SilverStripe\Core\Config\Configurable;
+use SilverStripe\Core\Convert;
+use SilverStripe\Core\Extensible;
+use SilverStripe\ORM\ArrayList;
+use SilverStripe\RedirectedURLs\Model\RedirectedURL;
+
+class RedirectedURLService
```
Comment (maxime-rainville) [inline src/Service/RedirectedURLService.php:15]: Maybe define an interface for this service and add some documentation for the class.

File: src/Service/RedirectedURLService.php [hunk 2]
```diff
@@ -0,0 +1,133 @@
...
+        return $response;
+    }
+
+    /**
+     * Converts an array of key value pairs to lowercase
+     *
+     * @param array $vars key value pairs
+     * @return array
+     */
+    protected function arrayToLowercase($vars)
```
Comment (maxime-rainville) [inline src/Service/RedirectedURLService.php:102]: Unless we explicitly expect people to extend this class and call this method, I would make it private to minimise our API surface.

File: src/Service/RedirectedURLService.php [hunk 3]
```diff
@@ -0,0 +1,133 @@
...
+                $result[strtolower($k)] = $this->arrayToLowercase($v);
+            } else {
+                $result[strtolower($k)] = strtolower($v);
+            }
+        }
+
+        return $result;
+    }
+
+    public function getRedirectCode($redirectedURL = null)
```
Comment (maxime-rainville) [inline src/Service/RedirectedURLService.php:117]: Add a doc block and define return type.

File: src/Extension/AssetStoreURLHandler.php [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+use SilverStripe\Control\HTTPResponse;
+use SilverStripe\Core\Extension;
+use SilverStripe\Core\Injector\Injector;
+use SilverStripe\RedirectedURLs\Model\RedirectedURL;
+use SilverStripe\RedirectedURLs\Service\RedirectedURLService;
+
+/**
+ * Class AssetStoreURLHandler
+ * @package SilverStripe\RedirectedURLs\Extension
+ *
```
Comment (maxime-rainville) [inline src/Extension/AssetStoreURLHandler.php:15]:
```suggestion
```

File: src/Extension/RedirectedURLHandler.php [hunk 5]
```diff
@@ -25,128 +27,19 @@
...
 class RedirectedURLHandler extends Extension
 {
-
-    /**
-     * Converts an array of key value pairs to lowercase
-     *
-     * @param array $vars key value pairs
-     * @return array
-     */
-    protected function arrayToLowercase($vars)
```
Comment (maxime-rainville) [inline src/Extension/RedirectedURLHandler.php:35]: That would be considered an API change.

File: src/Extension/RedirectedURLHandler.php [hunk 6]
```diff
@@ -25,128 +27,19 @@
...
-                $result[strtolower($k)] = $this->arrayToLowercase($v);
-            } else {
-                $result[strtolower($k)] = strtolower($v);
-            }
-        }
-
-        return $result;
-    }
-
-    protected function getRedirectCode($redirectedURL = false)
```
Comment (maxime-rainville) [inline src/Extension/RedirectedURLHandler.php:50]: That would be considered an API change.

**Review by maxime-rainville — CHANGES_REQUESTED** [review]
Haven't tested it locally. But at first glance it looks good.

Since the entire point of the PR is to redirect assets, I think we should have a test case that explicitly test that.

**Comment by oddnoc** [comment]
@madmatt Are you interested in pursuing this? Otherwise, I could take over the task.

**Comment by madmatt** [comment]
Hey @oddnoc , feel free to do so if you want! I am using a fork for a recent project which I really need to get rid of, but I just haven't found the time recently. If you end up getting this updated and merged in I'd love to know about it!

Thank you!

**Comment by n8-dev** [comment]
Since https://github.com/silverstripe/silverstripe-assets/pull/391 has been merged it would be great to get this merged in 🙏

**Comment by chrispenny** [comment]
Closing as we'll continue this in #86

**Pull request closed without merging**